hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -749,7 +753,7 @@ public Object put(Data key, Object value, long ttl) {
         Record record = getRecordOrNull(key, now, false);
         Object oldValue;
         if (record == null) {
-            oldValue = mapDataStore.load(key);
+            oldValue = loadFromStore ? mapDataStore.load(key) : null;
```

Comment:
Minor change at the end of the method. Can you move the duplication of ' saveIndex(record);' till after the if/else branch instead of repeating it in each branch?